planner/core/expression_rewriter.go

```diff
@@ -1259,11 +1259,29 @@ func (er *expressionRewriter) betweenToExpression(v *ast.BetweenExpr) {
 	if er.err != nil {
 		return
 	}
+
+	expr, lexp, rexp := er.ctxStack[stkLen-3], er.ctxStack[stkLen-2], er.ctxStack[stkLen-1]
+
+	if expression.AggCmpType(expr.GetType(), lexp.GetType(), rexp.GetType()) == types.ETString {
```

Comment:
How about add some comments for further maintain like `from MySQL Item_func_between::resolve_type`.